src/test/java/api/loans/CheckOutByBarcodeTests.java

```diff
@@ -163,6 +166,17 @@ public void canCheckOutUsingItemAndUserBarcode()
     assertThat(sessionRecord.getString("patronId"), is(steve.getId()));
     assertThat(sessionRecord.getString("loanId"), is(response.getId()));
     assertThat(sessionRecord.getString("actionType"), is("Check-out"));
+
+    assertTrue(loan.getJsonObject("item").containsKey(EFFECTIVE_CALL_NUMBER_COMPONENTS));
+
+    EffectiveCallNumberComponents callNumberComponents = loan
+      .getJsonObject("item")
+      .getJsonObject(EFFECTIVE_CALL_NUMBER_COMPONENTS)
+      .mapTo(EffectiveCallNumberComponents.class);
```

Comment:
I'd prefer we didn't use domain objects in assertions for API tests, this introduces a coupling between the internal implementation and the API tests.